pkg/eval/mods/file/file.go

```diff
@@ -124,3 +127,43 @@ func prclose(p vals.Pipe) error {
 func pwclose(p vals.Pipe) error {
 	return p.WriteEnd.Close()
 }
+
+//elvdoc:fn truncate
+func truncate(name string, size vals.Num) error {
+
+	switch size.(type) {
+	case int:
+		s := int64(size.(int))
+		err := os.Truncate(name, s)
+		if err != nil {
+			return err
+		} else {
+			return nil
+		}
+
+	case *big.Int:
+		if (size.(*big.Int)).IsInt64() {
+			s := (size.(*big.Int)).Int64()
+			err := os.Truncate(name, s)
```

Comment:
Don't duplicate this code. It's also far more verbose than necessary. This is a situation where a different pattern is clearer and less likely to contain errors:

```
var offset int64
switch size.(type) {
case ... // return early if an invalid value is detected otherwise assign the value to `offset`.
}

return os.Truncate(name, offset)
```

Reply:
Thank you, this is a great piece of advice, I will implement this here as well as in future commits. Thank you, Mr. @krader1961